spec/requests/security_activity_spec.rb

```diff
@@ -1,6 +1,19 @@
 RSpec.describe "security activities" do
   let(:user) { FactoryBot.create(:user) }
 
+  context "registering a new user" do
+    before { FactoryBot.create(:registration_state, :finished) }
+
+    it "records USER_CREATED events" do
+      # Stub the Registration State to sneak past redirect safeguards
+      registration_state_factory = RegistrationState.find("7216ddfe-d225-4d28-8989-36734bb4c2cd")
```

Comment:
Use a `let` for the registration state, and then use `registration_state.id` here, to avoid the magic UUID.

Reply:
ah right, I wasn't sure about the order of precidence here.
Let doesn't run until invoked right?  and we want the record to have been created in the test db so it can be found!

but have just swapped that over to your suggestion and seems to be passing, so looks like everything progresses in the right order.

Nice improvement, thank you!

Reply:
Resolved here: https://github.com/alphagov/govuk-account-manager-prototype/pull/571/commits/abb8daab5c3fe15a6bfdea2d38d83a9a9d81c384